src/main/java/hello/controller/IssueController.java

```diff
@@ -0,0 +1,183 @@
+            case 1:
+                result = session.run(Constants.MODIFIER_ORDER_FIX_NEO4J_QUERY).list();
+                for(Record record : result) {
+                    codeGenPostRequest.fileIds.add(record.get("file_id").asInt());
+                    fixIssue1(record.get("method_id").asInt(), record.get("modifiers").asString());
+                }
+
+                try {
+                    fixId = sendPostRequestCodeGen(Constants.CODEGEN_URL, new Gson().toJson(codeGenPostRequest)).getId();
+                } catch (CodeGenPostRequestException exception) {
+                    return new  Gson().toJson(new PostResponse(exception.getMessage()));
+                }
+
+                for(Record record : result) {
+                    fixesRepo.save(new Fixes(fixId, issueRepo.findById(issueId).get(), sandBoxUrl, Integer.toString(record.get("line").asInt()), record.get("file").asString(), Integer.toString(record.get("columnName").asInt()), false));
+                }
+
+                session.close();
+                driver.close();
+                return new Gson().toJson(new PostResponse(fixId, "200"));
+            case 2:
+                return "2";
+            case 3:
+                result = session.run(Constants.LAZY_INITIALIZATION_SYNCHRONIZATION_NEO4J_QUERY).list();
+                for(Record record : result) {
+                    codeGenPostRequest.fileIds.add(record.get("file_id").asInt());
+                    fixIssue3(record.get("method_id").asInt(), record.get("modifiers").asString());
+                }
+                try {
+                    fixId = sendPostRequestCodeGen(Constants.CODEGEN_URL, new Gson().toJson(codeGenPostRequest)).getId();
+                } catch (CodeGenPostRequestException exception) {
+                    return new Gson().toJson(new PostResponse(exception.getMessage()));
+                }
+                for(Record record : result) {
+                    fixesRepo.save(new Fixes(fixId, issueRepo.findById(issueId).get(), sandBoxUrl, Integer.toString(record.get("line").asInt()), record.get("file").asString(), Integer.toString(record.get("columnName").asInt()), false));
+                }
+                session.close();
+                driver.close();
+                return new Gson().toJson(new PostResponse(fixId, "200"));
+            default:
+                return "Invalid Issue Id";
+        }
+    }
+
+    @GetMapping(path="/fix/{fixId}", produces = "application/json")
+    public String getIssueStatus(@PathVariable(value="fixId") Integer fixId) {
+        try {
+            List<Fixes> fixesList = fixesRepo.findAllByFixId(fixId);
+
+            if (fixesList.isEmpty()) {
+                return new Gson().toJson(new GetResponse(fixId, "404"));
+            }
+
+            CodeGenGetResponse codeGenGetResponse = sendGetRequestCodeGen("http://codegen-cnu.ey.devfactory.com/api/codegen/status/" + fixId);
+
+            if (codeGenGetResponse.status.equals("Success")) {
```

Comment:
also handle the case when codegen fails due to some error.
Return an appropriate error message along with a status code.

Reply:
what is the response from codegen when it fails?